examples/web-socket-events/src/main/java/com/iland/app/App.java

```diff
@@ -0,0 +1,136 @@
+  }
+
+  /**
+   * Basic authentication using iland's Client object.
+   */
+  static void authentication() {
+    try {
+      apiClient.login(USERNAME, PASSWORD);
+    } catch (final OAuthException e) {
+      System.out.println(e.getMessage());
+    }
+  }
+
+  /**
+   * Lazily grab a org uuid from a user's inventory.
+   */
+  static String setCompanyIdAndGetOrgUuid() {
+    String orgUuid = "";
+    UserInventoryResponse userInventory =
+        userResource.getInventory(USERNAME, null);
+    for (UserCompanyInventoryResponse companyInventory : userInventory
+        .getInventory()) {
+      COMPANY_ID = companyInventory.getCompanyId();
+      if (companyInventory.getEntities() != null && !companyInventory
+          .getEntities().isEmpty()) {
+        List<UserInventoryEntityResponse> companies =
+            companyInventory.getEntities().get(IamEntityType.COMPANY);
+        for (UserInventoryEntityResponse company : companies) {
+          if (orgUuid.isEmpty()) {
+            orgUuid = company.getUuid();
+            break;
+          }
+        }
+      }
+    }
+    return orgUuid;
+  }
+
+  /**
+   * This function connects to a web socket and gets events based on the type
+   * we specify. We use two different examples of how to filter events. The
+   * first way is using a Set of EventTypes and the second way is to use
+   * a custom EventFilter. We need to keep the process alive thus the need for
+   * the while loop. Note if we want to not consume at some point in time
+   * we can use the unSubscribe() function in the SocketSubscription.
+   */
+  static void webSocketProcessEvents() {
+    final WebSocketClient webSocketClient =
+        apiClient.getEventWebSocket(COMPANY_ID);
+
+    final Set<EventType> vmTypes = new HashSet<>(Arrays.asList(vmEventTypes));
+     Consumer<EventResponse> vmEventConsumer = c -> log.info(String
+        .format("User %s initiated vm event %s for entity %s",
+            c.getInitiatedByUsername(), c.getType(), c.getEntityName()));
+    final SocketSubscription vmSubscription =
+        webSocketClient.consumeEvents(vmEventConsumer, vmTypes);
+
+    final Set<EventType> orgTypes = new HashSet<>(Arrays.asList(orgEventTypes));
+    final EventFilter orgEventFilter =
+        eventResponse -> orgTypes.contains(eventResponse.getType())
+            && eventResponse.getEntityUuid().equals(ORG_UUID);
+    final Consumer<EventResponse> orgEventConsumer = c -> log.info(String
+        .format("User %s initiated org event %s for entity %s",
+            c.getInitiatedByUsername(), c.getType(), c.getEntityName()));
+    final SocketSubscription orgSubscription =
+        webSocketClient.consumeEvents(orgEventConsumer, orgEventFilter);
+
+    // Keeping the process alive so the web socket can consume events.
+    // Note that there is no error handling for interrupted or connection errors
+    // and that you should add try/catch block to handle such cases.
+    while (true) {
```

Comment:
@nksprzak this is really not good. You're gonna keep the CPU very busy doing nothing. At a minimum, you need to call sleep within this loop.